### PR TITLE
ARROW-3723: [Plasma][Ruby] Add Ruby bindings of Plasma

### PR DESCRIPTION
**Pull request by shiro615**
It's Ruby bindings of Plasma.

### CODE REVIEW

File: ruby/red-plasma/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+## Install
+
+Install Plasma GLib before install Red Plasma. Use [packages.red-data-tools.org](https://github.com/red-data-tools/packages.red-data-tools.org) for installing Plasma GLib.
+
+Note that the Plasma GLib packages are "unofficial". "Official" packages will be released in the future.
+
+Install Red Plasma after you install Plasma GLib:
+
+```text
+% gem install red-plasma
```
Comment (shiro615) [inline ruby/red-plasma/README.md]: I've not registered `red-plasma` to RubyGems yet . We will register it until 0.12.0.

File: ruby/red-plasma/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+Red Plasma uses [Plasma GLib](https://github.com/apache/arrow/tree/master/c_glib/plasma-glib) and [gobject-introspection gem](https://rubygems.org/gems/gobject-introspection) to generate Ruby bindings of Plasma.
+
+Plasma GLib is a C wrapper for [Plasma C++](https://github.com/apache/arrow/tree/master/cpp/plasma). GObject Introspection can't use Plasma C++ directly. Plasma GLib is a bridge between Plasma C++ and GObject Introspection.
+
+gobject-introspection gem is a Ruby bindings of GObject Introspection. Red Plasma uses GObject Introspection via gobject-introspection gem.
+
+## Install
+
+Install Plasma GLib before install Red Plasma. Use [packages.red-data-tools.org](https://github.com/red-data-tools/packages.red-data-tools.org) for installing Plasma GLib.
```
Comment (shiro615) [inline ruby/red-plasma/README.md]: We’ve not create packages for Plasma and Plasma GLib. We will create them in the future.

File: ruby/red-plasma/lib/plasma/client.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module Plasma
+  class Client
+    alias_method :initialize_raw, :initialize
+    def initialize(socket_path)
+      socket_path = socket_path.to_path if socket_path.respond_to?(:to_path)
```
Comment (kou) [inline ruby/red-plasma/lib/plasma/client.rb]: GObject Introspection has ``(type filename)` annotation. We'll be able to remove this customization when we put the annotation to Plasma GLib and improve gobject-introspection gem to call `#to_path` for "filename" type argument automatically.

See also:

  * https://wiki.gnome.org/Projects/GObjectIntrospection/Annotations/
  * https://gitlab.gnome.org/GNOME/gtk/blob/master/gtk/gtkaccelmap.c#L666

Comment (kou) [reply ruby/red-plasma/lib/plasma/client.rb]: We can work on this later.

File: ruby/red-plasma/README.md (hunk 2 above)
Comment (kou) [reply ruby/red-plasma/README.md]: We'll release the official repository until 0.12.0. See #2879.
We'll update this when we merge #2879.

**Review by kou — APPROVED** [review]
+1
Can you check my comments?

File: ruby/red-plasma/lib/plasma/client.rb (hunk 3 above)
Comment (shiro615) [reply ruby/red-plasma/lib/plasma/client.rb]: I see. I'll try to work on this later.

File: ruby/red-plasma/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+plasma_store_server -m 1000000000 -s /tmp/plasma
+```
+
+Creating a Plasma client
+
+```ruby
+require "pathname"
+require "plasma"
+
+client = Plasma::Client.new(Pathname("/tmp/plasma"))
```
[the remark this answers is not in the document]
Comment (shiro615) [reply ruby/red-plasma/README.md]: I've removed `Pathname`.

**Comment by kou** [comment]
`ruby/red-plasma/lib/plasma/client.rb` doesn't have license header.
I'll add it and merge this.

**Comment by kou** [comment]
CI failure is unrelated.
I'll merge this.

**Pull request closed without merging**